### PR TITLE
Rewrite mkldnn conv bn fuse pass tester

### PR DESCRIPTION
**Pull request by HulekJakub**
### PR types
Others

### PR changes
Others

### Describe
Deleted old C++ conv activation mkldnn fuse pass test and updated python one

Currently in PaddlePaddle there are 3 unit-test frameworks - one written in C++ and 2 in Python.
Old fuse pass tests (C++) fuse_pass_tester.cc are checking only if graph has correct number of / type of vertices after fusion. It doesn't check data. So these tests pass even if accuracy drops.
New python framework is superior because it checks both graph fusion and accuracy after fusion.

This change adds dedicated test for onednn conv_bn_fuse_pass 
It also fixes issue when model is using mkldnn and convolution has bias. 

Previous pass (when executed with mkldnn and the convolution had bias) merged both layers into one convolution. The problem was that while doing this it did not scale the biases properly. This implementation solves it.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bn_fuse_pass.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (Silv3S) [inline python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bn_fuse_pass.py]:
```suggestion
# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```

File: python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bn_fuse_pass.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            return np.random.random(scale_shape).astype(np.float32)
+
+        def generate_bn_Bias():
+            return np.random.random(bias_shape).astype(np.float32)
+
+        def generate_bn_Mean():
+            return np.random.random(mean_shape).astype(np.float32)
+
+        def generate_bn_Var():
+            return np.random.random(var_shape).astype(np.float32)
```
Comment (Silv3S) [inline python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bn_fuse_pass.py]: maybe create one method with shape passed as argument

File: python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bn_fuse_pass.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            },
+            outputs=["batch_norm_Y"],
+        )
+        if has_bias:
+            program_config.weights["conv2d_bias"] = TensorConfig(
+                data_gen=partial(generate_conv2d_Bias)
+            )
+
+        print(conv2d_op.inputs)
+        print(program_config.weights)
```
Comment (Silv3S) [inline python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bn_fuse_pass.py]:
```suggestion
        print(conv2d_op.inputs)
        print(program_config.weights)
```

File: python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bn_fuse_pass.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+import hypothesis.strategies as st
+import numpy as np
+from auto_scan_test import PassAutoScanTest
+from program_config import OpConfig, ProgramConfig, TensorConfig
+
+
+class TestOneDNNConvBnFusePass(PassAutoScanTest):
+    def is_program_valid(self, program_config: ProgramConfig) -> bool:
+        return True
```
Comment (Silv3S) [inline python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bn_fuse_pass.py]:
```suggestion
    def is_program_valid(self, program_config: ProgramConfig) -> bool:
        return True
```
No need to overload this

File: paddle/fluid/framework/ir/conv_bn_fuse_pass.cc [hunk 5]
```diff
@@ -359,73 +359,95 @@ void ConvBNFusePass::ApplyImpl(ir::Graph* graph) const {
...
-          input_names.end();
-      if (has_bias && conv->Op()->Input("Bias").size() > 0) {
+      if (mkldnn_with_bias) {
         // reuse existing conv bias node
         auto conv_bias_names = conv->Op()->Input("Bias");
         PADDLE_ENFORCE_EQ(
             conv_bias_names.size(),
             1UL,
-            platform::errors::InvalidArgument("Find input var Bais error."));
+            platform::errors::InvalidArgument("Find input var Bias error."));
```
Comment (Silv3S) [inline paddle/fluid/framework/ir/conv_bn_fuse_pass.cc]:
```suggestion
            phi::errors::InvalidArgument("Find input var Bias error."));
```

File: paddle/fluid/framework/ir/conv_bn_fuse_pass.cc [hunk 6]
```diff
@@ -359,73 +359,95 @@ void ConvBNFusePass::ApplyImpl(ir::Graph* graph) const {
...
-                "phi::DenseTensor convolution bias(%d) and elementwise y(%d) "
-                "must have same dims.",
-                conv_bias_tensor->dims().size(),
-                eltwise_y_in_tensor->dims().size()));
-
-        auto eigen_conv_bias = EigenVector<float>::From(*conv_bias_tensor);
-        eigen_conv_bias += EigenVector<float>::From(*eltwise_y_in_tensor);
+        PADDLE_ENFORCE_EQ(conv_bias_tensor->dims(),
+                          bn_bias_tensor->dims(),
+                          platform::errors::InvalidArgument(
```
Comment (Silv3S) [inline paddle/fluid/framework/ir/conv_bn_fuse_pass.cc]:
```suggestion
                          phi::errors::InvalidArgument(
```

**Review by Silv3S — CHANGES_REQUESTED** [review]

**Review by Silv3S — DISMISSED** [review]

**Review by Silv3S — DISMISSED** [review]
LGTM

**Comment by HulekJakub** [comment]
@XieYunshen could you approve? I deleted obsolete test and fixed fuse pass.
 '::mutable_data()' was there already and I just moved it. I don't have access to DeviceContext in this class so I can not refactor it. Thank you

**Review by XieYunshen — DISMISSED** [review]
LGTM

**Comment by HulekJakub** [comment]
@YuanRisheng could you approve?
`::mutable_data()` was there already and I just moved it. I don't have access to DeviceContext in this class so I can not refactor it. Thank you

**Review by YuanRisheng — DISMISSED** [review]

**Comment by HulekJakub** [comment]
@Silv3S @YuanRisheng @XieYunshen I had to resolve conflicts, could you approve once again?

**Review by Silv3S — DISMISSED** [review]

**Review by Silv3S — APPROVED** [review]

**Comment by HulekJakub** [comment]
@YuanRisheng @XieYunshen I had to resolve conflicts, could you approve once again?

**Review by YuanRisheng — APPROVED** [review]

**Comment by HulekJakub** [comment]
@XieYunshen please approve changes in tests

**Review by XieYunshen — APPROVED** [review]

**Review by jczaja — APPROVED** [review]
LGTM

**Pull request merged**